### PR TITLE
[Fix](nereids) fix leading with left join without leftside condition

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
## Proposed changes

When left join with no edge in outer side, we should add outer side to minimal require of left tables

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yiguolei** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/CollectJoinConstraint.java [hunk 1]
```diff
@@ -162,6 +162,9 @@ private void collectJoinConstraintList(LeadingHint leading, Long leftHand, Long
                 }
             }
         }
+        if (minLeftHand == 0L) {
+            minLeftHand = leftHand;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/CollectJoinConstraint.java:166]: add test case

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Comment by morrySnow** [comment]
run feut

**Review by xzj7019 — APPROVED** [review]

**Comment by morrySnow** [comment]
run beut

**Pull request merged**